### PR TITLE
Separate serialization routines for `AttestationObject` and `MakeCredentialsResult`

### PR DESCRIPTION
**Pull request by jschanck**
The CTAP [authenticatorMakeCredential response](https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-errata-20220621.html#authenticatormakecredential-response-structure) structure includes a WebAuthn [attestation object](https://w3c.github.io/webauthn/#attestation-object) and additional (optional) fields. We previously identified the CTAP and WebAuthn structures because we didn't support the optional fields on the CTAP side. This PR separates the two structures, which will allow us to extend the CTAP structure.

We had previously manually implemented both `Serialize` and `Deserialize` for `AttestationObject`. Somewhat confusingly, the `Serialize` implementation produced WebAuthn-style maps with string keys ("fmt", "authData", "attStmt"), and the `Deserialize` implementation consumed CTAP-style maps with integer keys (1, 2, 3, respectively). The `Deserialize` implementation has been moved to `MakeCredentialsResult`.

Two bugs in the Deserialize implementation have been fixed. The new version
1) ignores unknown keys, and
2) does not accept a missing "attStmt (0x03)" field as a valid encoding of the "none" attestation format.

This PR also adds `#[derive(Deserialize)]` to `AttestationObject`. This will allow us to use authenticator-rs to parse WebAuthn attestation objects in Firefox, which we need for [Bug 1816519](https://bugzilla.mozilla.org/show_bug.cgi?id=1816519) and [Bug 1816520](https://bugzilla.mozilla.org/show_bug.cgi?id=1816520). (Why can we derive this Deserialize and not the one for `MakeCredentialsResult`? Serde cannot automatically map the integer keys to named struct fields in `MakeCredentialsResult`.) FWIW we manually implement Serialize to ensure that we produce CTAP canonical CBOR, and the derived Deserialize routine will accept non-canonical encodings. I think this is OK but I'm willing to change it.

Finally, because I was changing the definition of `MakeCredentialsResult`, I took the opportunity to remove the unused CTAP1 variants of  `RegisterResult` and `SignResult`.

### CODE REVIEW

File: src/ctap2/attestation.rs [hunk 1]
```diff
@@ -338,19 +342,31 @@ impl From<&[u8]> for Signature {
     }
 }
 
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug, PartialEq, Eq, Deserialize)]
+#[serde(tag = "fmt", content = "attStmt", rename_all = "lowercase")]
```
Comment (msirringhaus) [inline src/ctap2/attestation.rs:349]: Could you maybe add an explanatory comment here, what exactly this does? So one doesn't have to necessarily read the serde-docs to get a rough idea of whats happening.

Comment (msirringhaus) [reply src/ctap2/attestation.rs:349]: Ah, it's explained below. Nevermind then. Maybe point to that explanation, but I think it's fine to leave it as is.

Comment (jschanck) [reply src/ctap2/attestation.rs:349]: It is a bit confusing. I added a comment to AttestationStatement as well.

**Comment by jschanck** [comment]
When I went to incorporate this into Firefox, I found that I wanted a helper function to anonymize attestation objects from Windows Hello. So I'm tacking that onto this PR.

It also makes sense to use anonymize on `MakeCredentialsResult`, so I switched back to having an `AttestationObject` inside of `MakeCredentialsResult`.

**Review by mozkeeler — APPROVED** [review]
LGTM!

**Comment by jschanck** [comment]
For posterity: this also fixes a bug in the deserialization of authenticatorMakeCredential responses wherein we did not call `map.next_value()`, [here](https://searchfox.org/mozilla-central/rev/503938c13ef2dd174705dc0f6d0683ae43074ccc/third_party/rust/authenticator/src/ctap2/attestation.rs#528-530), after reading the attStmt key if the format was none. See also [Bug 1838077](https://bugzilla.mozilla.org/show_bug.cgi?id=1848077).

**Pull request merged**